application/src/main/java/org/togetherjava/tjbot/commands/github/GitHubReference.java

```diff
@@ -0,0 +1,169 @@
+            int id = Integer.parseInt(matcher.group(1));
+
+            try {
+                Optional<GHIssue> issue = findIssue(id);
+
+                if (issue.isPresent()) {
+                    embeds.add(generateReply(issue.get()));
+                }
+            } catch (IOException ex) {
+                throw new UncheckedIOException(ex);
+            }
+        }
+
+        List<List<MessageEmbed>> partition = ListUtils.partition(embeds, 10);
+        boolean first = true;
+        TextChannel textChannel = message.getTextChannel();
+
+        for (List<MessageEmbed> messageEmbeds : partition) {
+            if (first) {
+                message.replyEmbeds(messageEmbeds).mentionRepliedUser(false).queue();
+
+                first = false;
+            } else {
+                textChannel.sendMessageEmbeds(messageEmbeds).mentionRepliedUser(false).queue();
+            }
+        }
+    }
+
+    /**
+     * Generates the embed to reply with when someone references an issue
+     */
+    protected MessageEmbed generateReply(GHIssue issue) throws IOException {
+        return new EmbedBuilder()
+            .setColor(issue.getState() == GHIssueState.OPEN ? Color.green.getRGB()
+                    : Color.red.getRGB())
+            .setDescription(issue.getBody())
+            .setTitle("[#%d] %s".formatted(issue.getNumber(), issue.getTitle()),
+                    issue.getHtmlUrl().toString())
+            .setAuthor(issue.getUser().getName())
+            .setFooter("%s • %s".formatted(
+                    issue.getLabels()
+                        .stream()
+                        .map(GHLabel::getName)
+                        .collect(Collectors.joining(", ")),
+                    issue.getAssignees()
+                        .stream()
+                        .map(this::getUserNameOrFailAtRuntime)
+                        .collect(Collectors.joining(", "))))
+            .build();
+    }
+
+    // this is utterly stupid
+    /**
+     * Either properly gathers the name of a user or throws a UncheckedIOException
+     */
+    private String getUserNameOrFailAtRuntime(GHUser user) throws UncheckedIOException {
+        try {
+            return user.getName();
+        } catch (IOException ex) {
+            throw new UncheckedIOException(ex);
+        }
+    }
+
+    /**
+     * Safely (returning an optional instead of throwing an exception) looks through all of the
+     * given repositories for an issue/pr with the given id
+     */
+    protected Optional<GHIssue> findIssue(int id) {
+        return repositories.stream().map(repository -> {
+            try {
+                return Optional.of(repository.getIssue(id));
+            } catch (FileNotFoundException ignored) {
+                return Optional.<GHIssue>empty();
+            } catch (IOException ex) {
+                throw new UncheckedIOException(ex);
+            }
+        }).filter(Optional::isPresent).map(Optional::get).findAny();
```

Comment:
dont use `get`, use `orElseThrow` instead. `get` might get deprecated

Reply:
but I'm 100% certain that the thing isn't empty no?